src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -0,0 +1,231 @@
+
+    // insert a single row into testRun
+    if (BigQueryUtils.checkRowExists(
+        bigQueryClient, projectId, datasetName, testRunTableName, "id", testRunSummary.id)) {
+      logger.warn(
+          "A row with this id already exists in the "
+              + testRunTableName
+              + " table. Inserting a duplicate.");
+    }
+    TableId tableId = TableId.of(datasetName, testRunTableName);
+    InsertAllRequest insertRequest =
+        InsertAllRequest.newBuilder(tableId).addRow(buildTestRunRow(outputDirectory)).build();
+    BigQueryUtils.insertAllIntoBigQuery(bigQueryClient, insertRequest);
+
+    // insert into testScriptResults
+    tableId = TableId.of(datasetName, testScriptResultsTableName);
+    InsertAllRequest.Builder insertRequestBuilder = InsertAllRequest.newBuilder(tableId);
+    // Store test summaries in a Map with test names as keys
+    Map<String, TestScriptResult.TestScriptResultSummary> testScriptResultSummaries =
+        new ConcurrentHashMap<String, TestScriptResult.TestScriptResultSummary>();
+    testRunSummary.testScriptResultSummaries.stream()
+        .forEach(
+            runSummary -> testScriptResultSummaries.put(runSummary.testScriptName, runSummary));
+    // Loop through all test summaries
+    for (TestScriptSpecification testScriptSpecification : renderedTestConfiguration.testScripts) {
+      if (testScriptResultSummaries.containsKey(testScriptSpecification.name)) {
+        insertRequestBuilder.addRow(
+            buildTestScriptResultsRow(
+                testScriptSpecification,
+                testScriptResultSummaries.get(testScriptSpecification.name)));
+      } else {
+        logger.debug("Test name {} not found in result summaries.", testScriptSpecification.name);
+      }
+    }
+    BigQueryUtils.insertAllIntoBigQuery(bigQueryClient, insertRequestBuilder.build());
+
+    // insert into measurementCollection
+    if (measurementCollectionSummaries == null) {
+      logger.info("No measurement summaries found.");
+    } else {
+      tableId = TableId.of(datasetName, measurementCollectionTableName);
+      insertRequestBuilder = InsertAllRequest.newBuilder(tableId);
+      for (MeasurementCollectionScript.MeasurementResultSummary measurementResult :
+          measurementCollectionSummaries) {
+        insertRequestBuilder.addRow(buildMeasurementCollectionRow(measurementResult));
+      }
+      BigQueryUtils.insertAllIntoBigQuery(bigQueryClient, insertRequestBuilder.build());
+    }
+  }
+
+  /** Build a single row for each measurement collection. */
+  private Map<String, Object> buildMeasurementCollectionRow(
+      MeasurementCollectionScript.MeasurementResultSummary measurementResult) {
+    Map<String, Object> rowContent = new HashMap<>();
+
+    rowContent.put("testRun_id", testRunSummary.id);
+
+    rowContent.put("description", measurementResult.description);
+
+    rowContent.put("statistics_min", String.valueOf(measurementResult.statistics.min));
```

Comment:
Why are these string columns and not FLOAT64?

Reply:
They are FLOAT columns in the BQ dataset, just supplied as Strings to the insert method in the BQ client library, as recommended in the [Javadocs](https://googleapis.dev/java/google-cloud-bigquery/latest/com/google/cloud/bigquery/InsertAllRequest.Builder.html#addRow-java.util.Map-).